### ISSUE TITLE
Insufficient memory during prediction.

### ISSUE THREAD

**Issue by goodsave** [MIC-DKFZ/nnUNet#2846]
The content of planjson is as follows:

```
"plans_name": "nnUNetPlans",
    "original_median_spacing_after_transp": [
        1.5,
        1.5,
        1.5
    ],
    "original_median_shape_after_transp": [
        227,
        227,
        240
    ],
    "image_reader_writer": "NibabelIOWithReorient",
    "transpose_forward": [
        0,
        1,
        2
    ],
    "transpose_backward": [
        0,
        1,
        2
    ],

"3d_lowres": {
            "data_identifier": "nnUNetPlans_3d_lowres",
            "preprocessor_name": "DefaultPreprocessor",
            "batch_size": 2,
            "patch_size": [
                128,
                128,
                128
            ],
            "median_image_size_in_voxels": [
                196,
                196,
                206
            ],
            "spacing": [
                1.7389111114500002,
                1.7389111114500002,
                1.7389111114500002
            ],
            "normalization_schemes": [
                "CTNormalization"
            ],
            "use_mask_for_norm": [
                false
            ],
            "UNet_class_name": "PlainConvUNet",
            "UNet_base_num_features": 32,
            "n_conv_per_stage_encoder": [
                2,
                2,
                2,
                2,
                2,
                2
            ],
            "n_conv_per_stage_decoder": [
                2,
                2,
                2,
                2,
                2
            ],

```

But when I predict a CT with 268 layers and Z-spacing of 5, the program will first generate a predicted logits of size 25893465465 to store the resample predicted logits. (There are 25 labels on the right side of this model).

But my physical machine only has 32GB of memory, and the predicted loss of memory will cause OOM to occur due to the large amount of memory occupied by the predicted loss of bits. We attempted to change predicted w_logs to float16 and use the latest pytorch prediction to only partially alleviate the situation.

I would like to make a suggestion. Can we declare only one predicted logits with a size of [len (labels), z, y, x] in advance, and immediately resample it after predicting a patch (prediction=self, internal may be mirror_and_predict...), and then save it to the corresponding location in the predicted logits. This way, the total memory occupied during the entire prediction period will be less.

If modifying this method is laborious, an additional judgment can be added. If the size after resampling is too large, you can try splitting it from the Z-axis and predicting it in 2 or 3 times. Alternatively, a quick script can be provided to modify the size of the system's Swap memory. When the system memory is insufficient, Swap memory can be used to prevent the program from directly OOM or getting stuck.

Sincerely,

**Comment by goodsave** [comment]
[CT file](https://drive.google.com/file/d/1e6xDJafaBv4qp-uSJ2jzEX0Ig2-v7sLX/view?usp=sharing)

**Comment by vmiller987** [comment]
I would certainly be interested in this because my CT scans are slightly larger than average.

<img width="101" height="66" alt="Image" src="https://github.com/user-attachments/assets/d3b32148-b1f2-4593-b1b4-36cd14cf3141" />
